src/_repobee/command/repos.py

```diff
@@ -441,17 +463,44 @@ def _set_pull_ff_only(local_repos: List[plug.StudentRepo]) -> None:
         git.set_gitconfig_options(repo.path, {"pull.ff": "only"})
 
 
+def _with_output_paths(
+    repos: Iterable[plug.StudentRepo], directory_layout: DirectoryLayout
+) -> List[plug.StudentRepo]:
+    base_dir = pathlib.Path(".").resolve()
+    repo_path_func = _select_repo_path_function(directory_layout)
+    return [repo.with_path(repo_path_func(base_dir, repo)) for repo in repos]
+
+
+def _select_repo_path_function(
+    layout: DirectoryLayout,
+) -> Callable[[pathlib.Path, plug.StudentRepo], pathlib.Path]:
+    if layout == DirectoryLayout.FLAT:
+        return _flat_repo_path
+    elif layout == DirectoryLayout.NESTED_BY_TEAM:
+        return _nested_team_repo_path
+    else:
+        raise ValueError(f"unknown directory layout: {layout}")
```

Comment:
Codecov reports missing test. Maybe a good idea to test that wrong layouts fail properly? 

if anything to reach that sweet 100% coverage 😄

Reply:
Testing this case would be unnecessarily difficult as it's a "private" function of this module. I don't want to call the function directly in a test as I don't want to lock down its API, and it isn't possible to trigger this case without adding a new value to the enum (which requires some convoluted workarounds as enums are immutable). So, I wouldn't bother testing this case.